orca-igor/src/main/groovy/com/netflix/spinnaker/orca/igor/tasks/GetBuildPropertiesTask.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.igor.tasks;
+
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.igor.BuildService;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class GetBuildPropertiesTask extends RetryableIgorTask {
+  private final BuildService buildService;
```

Comment:
Does `@RequiredArgsConstructor` also do the dependency injection automatically?

Reply:
It doesn't directly, but if there's only a single constructor the Spring will automatically use it for autowiring.  (There is a way to ask lombok to annotate the constructor with the `onconstructor` argument, but I've never had to use it---I guess that would be necessary if someone added another constructor.)

Reply:
Oh gosh. I would love to make this kind of stuff explicit in the future - creating the constructor and adding the `@Autowired` annotation to it. It seems like an unnecessary mystery that doesn't buy us that much but could make it hard to spot bugs/changes in behavior. Especially as we're upgrading spring.

Reply:
Fair enough, I sometimes try too hard to avoid writing a few extra lines :)

I think it should should be robust to Spring upgrades as it's  explicitly documented by Spring in both [1.5](https://docs.spring.io/spring-boot/docs/1.5.19.RELEASE/reference/htmlsingle/#using-boot-spring-beans-and-dependency-injection) and [2](https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-spring-beans-and-dependency-injection.html) that a single constructor automatically gets autowired.  But it is true that it adds extra magic that a reader needs to understand (and maybe we should avoid interactions between annotations like this).